src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/EndToEndTests/NabovarselEndToEnd.cs

```diff
@@ -0,0 +1,292 @@
+            };
+
+            Uri uri = new Uri("/dibk/nabovarsel/instances", UriKind.Relative);
+
+            /* TEST */
+
+            HttpClient client = SetupUtil.GetTestClient(_factory, "dibk", "nabovarsel");
+            string token = PrincipalUtil.GetOrgToken("dibk");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            HttpResponseMessage response = await client.PostAsync(uri, formData);
+
+            response.EnsureSuccessStatusCode();
+
+            Assert.True(response.StatusCode == HttpStatusCode.Created);
+
+            Instance createdInstance = JsonConvert.DeserializeObject<Instance>(await response.Content.ReadAsStringAsync());
+
+            Assert.NotNull(createdInstance);
+            Assert.Equal(2, createdInstance.Data.Count);
+            #endregion
+
+            #region end user gets instance
+
+            // Reset token and client to end user
+            client = SetupUtil.GetTestClient(_factory, "dibk", "nabovarsel");
+            token = PrincipalUtil.GetToken(1337);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            string instancePath = "/dibk/nabovarsel/instances/" + createdInstance.Id;
+
+            HttpRequestMessage httpRequestMessage =
+            new HttpRequestMessage(HttpMethod.Get, instancePath);
+
+            response = await client.SendAsync(httpRequestMessage);
+            string responseContent = await response.Content.ReadAsStringAsync();
+            Instance instance = (Instance)JsonConvert.DeserializeObject(responseContent, typeof(Instance));
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("1337", instance.InstanceOwner.PartyId);
+            Assert.Equal("Task_1", instance.Process.CurrentTask.ElementId);
+            Assert.Equal(2, instance.Data.Count);
+            #endregion
+
+            #region end user gets application metadata
+
+            // Reset token and client to end user
+            client = SetupUtil.GetTestClient(_factory, "dibk", "nabovarsel");
+            token = PrincipalUtil.GetToken(1337);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            httpRequestMessage =
+            new HttpRequestMessage(HttpMethod.Get, "/dibk/nabovarsel/api/v1/applicationmetadata");
+
+            response = await client.SendAsync(httpRequestMessage);
+            responseContent = await response.Content.ReadAsStringAsync();
+            Application application = (Application)JsonConvert.DeserializeObject(responseContent, typeof(Application));
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            #endregion
+
+            #region Get Message DataElement
+
+            DataType dataType = application.DataTypes.FirstOrDefault(r => r.TaskId != null && r.TaskId.Equals(instance.Process.CurrentTask.ElementId));
```

Comment:
So, the return dataType here could either be the message or the dataElement describe with a guid.. 
Now we know the message is the first element in the list, but what are we trying to obtain here and should we be more specific about which dataType we are trying to find?

Reply:
I have tried to explain it better in the test now.